From host sync docs with examples (#458)
loft-sh/vcluster-docs: vcluster/configure/vcluster-yaml/sync/from-host/secrets.mdx

import ProAdmonition from '../../../../_partials/admonitions/pro-admonition.mdx'
import FromHostSecretExample from '../../../../_fragments/sync-from-host-secret-example.mdx'

By default, this is turned off.

:::info No need to configure RBAC
vCluster automatically adds the required cluster RBAC permissions for retrieving the secrets and syncing the resources from the host to the virtual cluster.
:::

Enabling this allows you to sync Secrets from the specified namespaces in the host to the specified namespaces in the vCluster.

It is also possible to modify the name of the synced resource in the virtual cluster.

There is no option to sync from all namespaces in the host.

Sync is one-directional, from host to virtual. If you modify an object in the host, vCluster syncs the change to virtual object. When you delete a virtual object, vCluster re-creates it if the host object still exist.

When you delete a host object, vCluster deletes the corresponding virtual object.

You can use synced Secrets in your workloads as an environment variables source or a volume.

It is not possible to sync Secrets that were already synced from virtual to host, they are skipped by vCluster.

> Namespaces in the virtual cluster are created automatically during the sync (if they do not exist already).

Reviewer: Same 2 comments as configMaps page

---
title: Secrets
sidebar_label: secrets
sidebar_position: 11
description: Configuration for ...
---

import EnableSwitch from '../../../../_partials/config/sync/fromHost/secrets.mdx'
import ProAdmonition from '../../../../_partials/admonitions/pro-admonition.mdx'
import FromHostSecretExample from '../../../../_fragments/sync-from-host-secret-example.mdx'

By default, this is turned off.

:::info No need to configure RBAC
vCluster automatically adds the required cluster RBAC permissions for retrieving the Secrets and syncing the resources from the host to the virtual cluster.
:::

> Enabling this allows you to sync Secrets from the specified namespaces in the host to the specified namespaces in the vCluster.

Reviewer 2: ```suggestion
Enabling sync from host allows you to sync Secrets from the specified namespaces in the host cluster to the specified namespaces in a virtual cluster.

Here is how the required configuration in `vcluster.yam` looks like:
```

> ```yaml title="vcluster.yaml"

Reviewer 2: ```suggestion
```yaml title="configure secret sync from host"
```

sync:
  fromHost:
    secrets:
      enabled: true
      selector:
        mappings:
          # syncs all Secrets from "foo" namespace

>           # to the "bar" namespace in vCluster. Secrets names are unchanged.

Reviewer 2: ```suggestion
          # to the "bar" namespace in a virtual cluster. Secrets names are unchanged.
```

          "foo/*": "bar/*"
```

> 

Reviewer 2: Would be good to add a short paragraph, something like:
Here are a few things to remember when configuring the sync.

- It is also possible to modify the name of the synced resource in the virtual cluster.
- There is no option to sync from all namespaces in the host.
- Sync is one-directional, from host to virtual. If you modify an object in the host, vCluster syncs the change to virtual object.
- When you delete a virtual object, vCluster re-creates it if the host object still exist.
- When you delete a host object, vCluster deletes the corresponding virtual object.

> It is not possible to sync Secrets that were already synced from virtual to host, they are skipped by vCluster.

Reviewer 2: Those should be bullet points too.

Namespaces in the virtual cluster are created automatically during the sync (if they do not exist already).

You can use synced Secrets in your workloads as an environment variables source or a volume.

> ## Prerequisites

Reviewer 2: Could we add `BasePrerequisites` partial like in the other guides? Eg here: https://github.com/loft-sh/vcluster-docs/blob/aba0e93a803729fc32947088c21173a7f71a7554/vcluster/configure/vcluster-yaml/experimental/native-sleep-mode.mdx?plain=1#L93

All the specified namespaces have to exist in the host at the vCluster startup.

## Sync all Secrets from host namespace

> To sync all Secrets from a given namespace in the host to the given namespace in the virtual cluster, “namespace/*” wildcard can be used, e.g.:

Reviewer 2: We try to avoid passive voice
```suggestion
To sync all Secrets from a given namespace in the host to the given namespace in the virtual cluster, use “namespace/*” wildcard, e.g.:
```

> ```yaml

Reviewer 2: This needs title

sync:
  fromHost:
    secrets:
      enabled: true
      selector:
        mappings:
          # syncs all Secrets from "foo" namespace

>           # to the "bar" namespace in vCluster. Secrets names are unchanged.

Reviewer 2: ```suggestion
          # to the "bar" namespace in a virtual cluster. Secrets names are unchanged.
```

          "foo/*": "bar/*"
```

## Sync only specific Secrets from host namespace
To sync only specific Secrets from namespaces, you need to provide `namespace/name` as the key and value:

> ```yaml

Reviewer 2: This needs title

sync:
  fromHost:
    secrets:
      enabled: true
      selector:
        mappings:
          # syncs Secret named "cm-name" from "foo" host namespace
          # to the "bar" namespace in virtual.
          "foo/cm-name": "bar/cm-name"
```

> ## Sync all Secrets from vCluster's host namespace

Reviewer 2: ```suggestion
## Sync all Secrets from virtual cluster's host namespace
```

> There is also a handy syntax to sync all Secrets from vCluster’s own host namespace to the virtual namespace. As vCluster’s namespace is not always known upfront (e.g. when vCluster is created by the platform), `""` (empty string) is treated as “vCluster’s own host namespace”.

Reviewer 2: ```suggestion
There is also a handy syntax to sync all Secrets from virtual cluster's own host namespace to the virtual namespace. As virtual cluster’s namespace is not always known upfront (e.g. when a virtual cluster is created by the platform), `""` (empty string) is treated as “virtual cluster’s own host namespace”.
```

> ```yaml

Reviewer 2: This needs title

sync:
  fromHost:
    secrets:
      enabled: true
      selector:
        mappings:

>           # syncs all Secrets from vCluster's host namespace

Reviewer 2: ```suggestion
          # syncs all Secrets from virtual cluster's host namespace
```

          # to "my-virtual" namespace in vCluster.
          "": "my-virtual"
```

> ## Sync specific Secret from vCluster's host namespace

Reviewer 2: ```suggestion
## Sync specific Secret from virtual cluster's host namespace
```

> you can also specify only a few Secrets from vCluster’s own host namespace this way:

Reviewer 2: ```suggestion
you can also specify only a few Secrets from virtual cluster’s own host namespace this way:
```

> ```yaml

Reviewer 2: This needs title

sync:
  fromHost:
    secrets:
      enabled: true
      selector:
        mappings:

>           # syncs Secret named "my-cm" from vCluster's host namespace

Reviewer 2: ```suggestion
          # syncs Secret named "my-cm" from virtual cluster's host namespace
```

>           # to "my-virtual-namespace" in vCluster.

Reviewer 2: ```suggestion
          # to "my-virtual-namespace" in a virtual cluster.
```

          "/my-cm": "my-virtual/my-cm"
```

## Modify synced Secret namespace and name in the virtual cluster
It’s also possible to modify Secret name during the sync:

```yaml
sync:
  fromHost:
    secrets:
      enabled: true
      selector:
        mappings:
          # syncs "config" Secret from "cert-manager" namespace in the host

>           # as "my-config" in "my-virtual" namespace in vCluster.

Reviewer 2: ```suggestion
          # as "my-config" in "my-virtual" namespace in a virtual cluster.
```

          "cert-manager/config": "my-virtual/my-config"
```

> ## Pro patches

Reviewer 2: ```suggestion
## Patches
```

<ProAdmonition/>

You can specify `reverseExpression` in the `sync.fromHost.secrets.patches` .

They are applied on the host object and appear in the virtual object.

`expressions` have no effect.

So, for the following `vcluster.yaml` :

> ```yaml

Reviewer 2: This needs title
